roles/backend_setup/README.md

```diff
@@ -0,0 +1,153 @@
+| glusterfs_infra_lv_thicklvsize || UNDEF | Optional. Needed only if thick volume has to be created. Include the unit [G\|M\|K]|
+| glusterfs_infra_mount_devices | | UNDEF | This is a dictionary with mount values. path, and lv are the keys. |
+| glusterfs_infra_ssd_disk | | UNDEF | SSD disk for cache setup, specific to HC setups. Should be absolute path. e.g /dev/sdc |
+| glusterfs_infra_lv_cachelvname | | glusterfs_ssd_cache | Optional varialbe, if omitted glusterfs_ssd_cache is used by default. |
+| glusterfs_infra_lv_cachelvsize | | UNDEF | Size of the cache logical volume. Used only while setting up cache. |
+| glusterfs_infra_lv_cachemetalvname | | UNDEF | Optional. Cache metadata volume name. |
+| glusterfs_infra_lv_cachemetalvsize | | UNDEF | Optional. Cache metadata volume size. |
+| glusterfs_infra_cachemode | | writethrough | Optional. If omitted writethrough is used. |
+
+
+#### VDO Variable
+------------
+If the backend disk has to be configured with VDO the variable glusterfs_infra_vdo has to be defined.
+
+| Name                     |Choices| Default value         | Comments                          |
+|--------------------------|-------|-----------------------|-----------------------------------|
+| glusterfs_infra_vdo || UNDEF | Mandatory argument if vdo has to be setup. Key/Value pairs have to be given. name and device are the keys, see examples for syntax. |
+
+```
+For Example:
+glusterfs_infra_vdo:
+   - { name: 'hc_vdo_1', device: '/dev/vdb' }
+   - { name: 'hc_vdo_2', device: '/dev/vdc' }
+   - { name: 'hc_vdo_3', device: '/dev/vdd' }
+```
+
+#### Logical Volume variable
+-----------------------
+| Name                     |Choices| Default value         | Comments                          |
+|--------------------------|-------|-----------------------|-----------------------------------|
+| glusterfs_infra_lv_logicalvols || UNDEF | This is a list of hash/dictionary variables, with keys, lvname and lvsize. See below for example. |
+
+```
+For Example:
+glusterfs_infra_lv_logicalvols:
+   - { lvname: 'hc_images', lvsize: '500G' }
+   - { lvname: 'hc_data', lvsize: '500G' }
+```
+
+#### Variables for mounting the filesystem
+-----------------------------------------
+| Name                     |Choices| Default value         | Comments                          |
+|--------------------------|-------|-----------------------|-----------------------------------|
+| glusterfs_infra_mount_devices | | UNDEF | This is a dictionary with mount values. path, and lv are the keys. |
+
+```
+For example:
+glusterfs_infra_mount_devices:
+        - { path: '/mnt/thinv', lv: <lvname> }
+        - { path: '/mnt/thicklv', lv: "{{ glusterfs_infra_lv_thicklvname }}" }
+```
+
+
+Example Playbook
+----------------
+
+Configure the ports and services related to GlusterFS, create logical volumes and mount them.
+
+
+```
```

Comment:
Adding ```yaml will help highlight the yaml syntax

Reply:
Same as above, it is not actually needed.